src/Generator/MethodGenerator.php

```diff
@@ -29,6 +30,8 @@ class MethodGenerator extends AbstractMemberGenerator
 
     private bool $returnsReference = false;
 
+    private bool $sortParametersOnSet = true;
```

Comment:
Why is this required?

Reply:
`setParameters` calls `setParameter` in a loop and `setParameter` calls `sortParameters`. Having `sortParameters` called multiple times seems a little inefficient, so `$sortParametersOnSet` gets set to `false` at the start of `setParameters` to avoid needless calls.

Reply:
@willjones-stratagem consider that `laminas/laminas-code` is not designed for runtime usage anyway: only for building sources.

I suggest removing any state mutation, and re-performing the operation, even if inefficient at first :-)

Reply:
@Ocramius fair shout. I'll update that.
That said, the project I'm working on at the moment will probably end up using `laminas/laminas-code` as part of certain runtime functionality so some overall efficiency savings in [5.x.x](https://github.com/laminas/laminas-code/issues/69) wouldn't be a bad thing

Reply:
That is alright, but I'd like to prevent state mutation as much as possible.

An exception thrown during a flag switch would cause a mess, for example, and subclassing makes this much more complex too.

An alternate design will be needed, if this performance impact is relevant in future.

Reply:
Very good point. The PR's now been updated.